### ISSUE TITLE
Unable to load linter-htmlhint

### ISSUE THREAD

**Issue by chriswelchgb** [AtomLinter/linter-htmlhint#129]
Hi, I've been cracking at this a couple hours and have not been able to figure it out. I've been able to get linter-jshint working, but still unable to get linter-htmlhint working. Here's what I've tried:

Installed the most recent linter and linter-htmlhint using the Atom package install screen
- Properly configured the path in the config file:
  
  ``` cson
  "linter-htmlhint":
      executablePath: "/Users/[my-username]/.atom/packages/linter-htmlhint/node_modules/.bin/htmlhint"
  ```
  
  -> this also showed up in the settings screen of the package, so I'm assuming it's reading the file correctly
- Created an .htmlhintrc file, and configured it.
- Restarted Atom (1.8.0)

At this point, tried to muck up an HTML file to see if linter-htmlhint would work, but no dice.

So then I:
- Used NPM to install htmlhint globally confirmed that my config file was working and htmlhint was validating as expected.
- Tested the executable within the executablePath using the same config and HTML file and got similar results.
- Did the uninstall / reinstall process a couple of times to make sure I wasn't doing something

Any thoughts?

As a point of reference, I ran the first process with linter-jshint and it worked fine, so I'm expecting similar results from htmlhint.

**Comment by chriswelchgb** [comment]
All - I've made progress with this issue since reporting it, however it is still not resolved the issue.

I had installed the language-liquid package, and I noticed that by default all .html files were being parsed by language-liquid instead of language-html. I switched the parser to HTML, and boom! I started seeing some life from linter-htmlhint, however it still does not work. I get this error now:

SyntaxError: Unexpected token d
    at Object.parse (native)
    at /Users/XXX/.atom/packages/linter-htmlhint/lib/index.js:33:21

**Comment by Arcanemagus** [comment]
Looks like that error is coming from attempting to parse your `.htmlhintrc` file, can you open Atom's developer tools, locate the `linter-htmlhint` files and place a breakpoint [here](https://github.com/AtomLinter/linter-htmlhint/blob/v1.2.1/lib/index.js#L33), and see if `conf` matches your file?

**Comment by chriswelchgb** [comment]
I was able to set the breakpoint, and it was the right file.  I was able to get it working, by going through it line by line and I'm not quite sure how. I didn't ultimately change any of the property names.

Coming out of this experience, I'd like to suggest a few documentation changes:
* I know this really should be an htmlhint ask, but I would have LOVED it if you guys provided a sample .htmlhintrc file with ALL settings in your setup instructions?  The one on the htmlhint site only has a couple of properties provided. Then we can just copy/paste and be off to the races.
* Mention the typical install locations of the htmlhint executable on Mac and Windows? (this took me 30 minutes to figure out)
* Mention to check what parser your files are using because HTMLHint will only work when the parser is set to HTML.

Thanks!
Chris

**Comment by Arcanemagus** [comment]
PR's fixing documentation are always welcome 😉.

I do what I can to keep the linter working, but this project doesn't seem to have an active maintainer currently. I don't actually use this project myself.

I'm not sure what you mean by "Parser" as HTMLHint doesn't seem to have an option for an alternative parser. Do you mean the language setting on the file?

**Comment by johnwebbcole** [comment]
I usually start with the [htmlhint](htmlhint.com) site and select all the rules, except for a few, then use that as my `.htmlhint.rc` file.  The `src/rules` directory in the htmlhint source, isn't too different from the website.

I'm not opposed to adding more info in the docs, but these locations seem to change quite a bit... atom moves things around a lot, and sometimes the local project htmlhint gets used.

I think #125 should make it obvious what scopes are currently being used (and allow you to add/remove scopes).  Do you think further documentation is needed?

**Comment by Arcanemagus** [comment]
#125 has been published in v1.3.0.

**Comment by chriswelchgb** [comment]
Arcanemagus - thanks for your continued commitment to this project!

When I was saying "parser" I was referring to the language package within Atom that color codes the file. If the user's file is not using language-html (for example, using plaintext or HTML Liquid), then HTML Hint will not work. That may seem obvious to check, but it was not obvious to me as I was troubleshooting.

John - thanks for jumping in here as well! Maybe just simply pointing out where you get your starter htmlhintrc file on the htmlhint github would be enough. Again, I was able to setup this plugin, but I had no background in htmlhint or Atom and so this one package took me over 4 hours to get working. My goal here is suggest ways to help others get it working in much less time.

As far as the location of the executable, even if it changes, I think you're doing a service to us novices by at least providing a hint (The executable should be located in XXX as of version 1.8, but it may change), will help shave off some time. This is the approach linter-JSHint takes, and ultimately how I found htmlhint.
